ui/ui.go

```
@@ -96,29 +96,52 @@ func (ui *WriterUI) PrintTable(table Table) {
 	}
 }
 
-func (ui *WriterUI) AskForText(label string) (string, error) {
+func (ui *WriterUI) AskForText(opts TextOpts) (string, error) {
 	var text string
 
-	err := interact.NewInteraction(label).Resolve(&text)
-	if err != nil {
-		return "", fmt.Errorf("Asking for text: %s", err)
+	for {
+		text = opts.Default
+		err := interact.NewInteraction(opts.Label).Resolve(&text)
+		if err != nil {
+			return "", fmt.Errorf("Asking for text: %s", err)
+		}
+
+		if opts.ValidateFunc != nil {
+			isValid, err := opts.ValidateFunc(text)
+			if isValid {
+				break
+			}
+			ui.PrintErrorBlock(err.Error())
```

shouldnt print anything in these functions (especially a "block" since block in this context means primary content). if err is encountered, we should probably just return an error from this method.

---

Okay so you mean we will continue in the `for` loop only when it's invalid and err is `nil` like when `validateFunc` (`return false, "", nil`)

---

But I believe what we discussed yesterday is not to return until we get the valid input if we return from here with an error then should we consider default value or we should return empty string here and zero in case of `AskForChoice` ?

---

In my opinion, maybe we should again prompt to the user to take input when it entered an invalid input instead of exiting with an error